hack/make/.go-compile-test-dir

```diff
@@ -23,9 +23,13 @@ if [ "$BUILDFLAGS_FILE" ]; then
 	readarray -t BUILDFLAGS < "$BUILDFLAGS_FILE"
 fi
 
+if [[ "`go version`" =~ "gccgo" ]]; then
+   GCCGOFLAGS="-gccgoflags= -lpthread -ldl "
```

Comment:
Can you please indent this with tabs like the rest of the file, and make sure to force it to be empty beforehand to make sure we don't have side effects?  Also, can you swap the backticks for `"$(go version)"` to match the other places we shell out as well?